Error gettting module working directory for multi-module project in Idea

**Context**
 - Plugin: Infinitest for IntelliJ
 - Plugin Version: 5.3.2-SNAPSHOT (tested as of https://github.com/infinitest/infinitest/pull/334/commits/0ecc0ffba5d8f33f9f10066b32d4a77d69857784 from #334)
 - IDE: IntelliJ 2022.1.4

**Describe the bug**
Infinitest throws exception in multi-module project on its initial rebuild. It seems that Infinitest is trying to execute in the non-existing Idea (sub)directory:
> IOException: Cannot run program "/lib/jvm/java-17-openjdk/bin/java" (in directory "/home/foobar/Code/ArchUnit-Examples/.idea/modules/example-junit4"): error=2, No such file or directory

Idea 2022.1.4 doesn't created any subdirectory in the root directory `ArchUnit-Examples/.idea`.

**To Reproduce**
1. Import https://github.com/TNG/ArchUnit-Examples into Idea
2. Rebuild the project (e.g. CTRL-F9)

**Expected behavior**
Tests in all the modules are executed.

**Additional info**

<details><summary>The stacktrace (click to expand)</summary>

```
java.lang.RuntimeException: java.io.IOException: Failed to start runner process with configuration
  Directory: /home/foobar/Code/ArchUnit-Examples/.idea/modules/example-junit4
  Environment:
    ...
  Command: [/lib/jvm/java-17-openjdk/bin/java, -mx256m, -Djava.system.class.loader=org.infinitest.classloader.ClassPathFileClassLoader, -Dorg.infinitest.classloader.classPathFile=/tmp/infinitest-10292390519896297138.classpath, org.infinitest.testrunner.TestRunnerProcess, org.infinitest.testrunner.DefaultRunner, 45441]
	at org.infinitest.testrunner.queue.QueueConsumer.startProcessing(QueueConsumer.java:71)
	at org.infinitest.testrunner.queue.QueueConsumer.push(QueueConsumer.java:62)
	at org.infinitest.testrunner.MultiProcessRunner.runTests(MultiProcessRunner.java:69)
	at org.infinitest.DefaultInfinitestCore.runTests(DefaultInfinitestCore.java:233)
	at org.infinitest.DefaultInfinitestCore.runOptimizedTestSet(DefaultInfinitestCore.java:126)
	at org.infinitest.DefaultInfinitestCore.update(DefaultInfinitestCore.java:82)
	at org.infinitest.DefaultInfinitestCore.update(DefaultInfinitestCore.java:92)
	at org.infinitest.intellij.idea.IdeaCompilationListener.doRunTests(IdeaCompilationListener.java:74)
	at org.infinitest.intellij.idea.IdeaCompilationListener.finished(IdeaCompilationListener.java:56)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeMethod(MessageBusImpl.java:642)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeListener(MessageBusImpl.java:617)
	at com.intellij.util.messages.impl.MessageBusImpl.deliverMessage(MessageBusImpl.java:417)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:391)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:373)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:34)
	at com.intellij.util.messages.impl.MessageBusImpl$MessagePublisher.invoke(MessageBusImpl.java:180)
	at com.sun.proxy.$Proxy116.finished(Unknown Source)
	at com.intellij.task.impl.ProjectTaskManagerImpl$ResultConsumer.lambda$notify$1(ProjectTaskManagerImpl.java:370)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:215)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:197)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:511)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:69)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:112)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:42)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:803)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.io.IOException: Failed to start runner process with configuration
  Directory: /home/foobar/Code/ArchUnit-Examples/.idea/modules/example-junit4
  Environment:
    ...
  Command: [/lib/jvm/java-17-openjdk/bin/java, -mx256m, -Djava.system.class.loader=org.infinitest.classloader.ClassPathFileClassLoader, -Dorg.infinitest.classloader.classPathFile=/tmp/infinitest-10292390519896297138.classpath, org.infinitest.testrunner.TestRunnerProcess, org.infinitest.testrunner.DefaultRunner, 45441]
	at org.infinitest.testrunner.process.NativeConnectionFactory.startProcess(NativeConnectionFactory.java:80)
	at org.infinitest.testrunner.process.NativeConnectionFactory.getConnection(NativeConnectionFactory.java:59)
	at org.infinitest.testrunner.TestQueueProcessor.establishConnection(TestQueueProcessor.java:67)
	at org.infinitest.testrunner.TestQueueProcessor.<init>(TestQueueProcessor.java:50)
	at org.infinitest.testrunner.MultiProcessRunner$1.createQueueProcessor(MultiProcessRunner.java:55)
	at org.infinitest.testrunner.queue.QueueConsumer.startProcessing(QueueConsumer.java:67)
	... 48 more
Caused by: java.io.IOException: Cannot run program "/lib/jvm/java-17-openjdk/bin/java" (in directory "/home/foobar/Code/ArchUnit-Examples/.idea/modules/example-junit4"): error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at org.infinitest.testrunner.process.NativeConnectionFactory.startProcess(NativeConnectionFactory.java:77)
	... 53 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 55 more
```

</details>

--- a/infinitest-intellij/src/main/java/org/infinitest/intellij/idea/IdeaModuleSettings.java
+++ b/infinitest-intellij/src/main/java/org/infinitest/intellij/idea/IdeaModuleSettings.java
@@ -40,6 +40,8 @@ import org.infinitest.intellij.InfinitestJarsLocator;
 import org.infinitest.intellij.ModuleSettings;
 import org.jetbrains.annotations.Nullable;
 
+import com.intellij.execution.CommonProgramRunConfigurationParameters;
+import com.intellij.execution.util.ProgramParametersConfigurator;
 import com.intellij.ide.plugins.IdeaPluginDescriptor;
 import com.intellij.ide.plugins.PluginManager;
 import com.intellij.openapi.compiler.CompilerPaths;
@@ -149,7 +151,12 @@ public class IdeaModuleSettings implements ModuleSettings {
 	}
 
 	private File getWorkingDirectory() {
-		return new File(module.getModuleFilePath()).getParentFile();
+		CommonProgramRunConfigurationParameters configuration = new InfinitestRunConfigurationParameters();
+		
+		ProgramParametersConfigurator configurator = new ProgramParametersConfigurator();
+		String workingDir = configurator.getWorkingDir(configuration, module.getProject(), module);
+		
+		return new File(workingDir);
 	}
 
 	/**

--- a/infinitest-intellij/src/main/java/org/infinitest/intellij/idea/InfinitestRunConfigurationParameters.java
+++ b/infinitest-intellij/src/main/java/org/infinitest/intellij/idea/InfinitestRunConfigurationParameters.java
@@ -1,0 +1,95 @@
+/*
+ * Infinitest, a Continuous Test Runner.
+ *
+ * Copyright (C) 2010-2013
+ * "Ben Rady" <benrady@gmail.com>,
+ * "Rod Coffin" <rfciii@gmail.com>,
+ * "Ryan Breidenbach" <ryan.breidenbach@gmail.com>
+ * "David Gageot" <david@gageot.net>, et al.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.infinitest.intellij.idea;
+
+import java.util.Map;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.intellij.execution.CommonProgramRunConfigurationParameters;
+import com.intellij.openapi.project.Project;
+
+/**
+ * ProgramParametersConfigurator.getWorkingDir only calls getWorkingDirectory and we want this to return null
+ * The interface is implemented but it does not seem to be necessary in the current version of IDEA
+ * 
+ * In a future version it might be good to actually implement a run configuration and let the user configure the
+ * runner from there
+ */
+public class InfinitestRunConfigurationParameters implements CommonProgramRunConfigurationParameters {
+	private String programParameters;
+	private String workingDirectory;
+	private Map<String, String> envs;
+	private boolean passParentEnvs;
+	
+	@Override
+	public Project getProject() {
+		return null;
+	}
+
+	@Override
+	public void setProgramParameters(@Nullable String value) {
+		this.programParameters = value;
+	}
+
+	@Override
+	public @Nullable String getProgramParameters() {
+		return programParameters;
+	}
+
+	@Override
+	public void setWorkingDirectory(@Nullable String value) {
+		this.workingDirectory = value;
+	}
+
+	@Override
+	public @Nullable String getWorkingDirectory() {
+		return workingDirectory;
+	}
+
+	@Override
+	public void setEnvs(@NotNull Map<String, String> envs) {
+		this.envs = envs;
+	}
+
+	@Override
+	public @NotNull Map<String, String> getEnvs() {
+		return envs;
+	}
+
+	@Override
+	public void setPassParentEnvs(boolean passParentEnvs) {
+		this.passParentEnvs = passParentEnvs;
+	}
+
+	@Override
+	public boolean isPassParentEnvs() {
+		return passParentEnvs;
+	}
+}

